[daily_ot/regression] Deal with license check out errors

Currently, if the license check out for the simulation tool fails for the daily sanity checks, such runs enter as failures into the results summary. This makes our summary (currently at https://reports.opentitan.org/hw/top_earlgrey/dv/summary.html) look worse than it actually is. For example, AES passes less than 35% of the tests, and we get the following errors on the [AES Simulation Results](https://reports.opentitan.org/hw/ip/aes/dv/latest/results.html) page:
```
xrun(64): 18.09-s011: (c) Copyright 1995-2019 Cadence Design Systems, Inc.

Trying to check out license...
	Xcelium_Single_Core 18.00 - Failed
	Xcelium_Single_Core_Legacy 18.00 - Failed
	Xcelium_Safety 18.00 - Failed
	Xcelium_Multi_Core 18.00 - Failed
	Xcelium_For_Partners 18.00 - Failed
xmsim: *F,NOLICN: Unable to checkout license for the simulation. 'lic_error -5'.
make: *** [/usr/local/google/home/weicai/daily_ot/hw/dv/data/sim.mk:71: simulate] Error 2
```

I think we should try to catch those errors and for example repeat the test until we actually get a license, or (as a quick workaround) not update the results if we get such errors. Especially as we move towards publishing more results more prominently (#1695), such things may give a bad first impression.

--- a/hw/dv/data/xcelium/xcelium.hjson
+++ b/hw/dv/data/xcelium/xcelium.hjson
@@ -9,7 +9,8 @@
   // {tool_srcs_dir} before running the simulation.
   tool_srcs:  ["{proj_root}/hw/dv/tools/xcelium/*"]
 
-  build_opts: [" -elaborate -64bit -access +r -sv",
+  build_opts: ["-elaborate -64bit -access +r -sv",
+               "-licqueue",
                "-messages -errormax 50",
                "-timescale 1ns/1ps",
                "-f {sv_flist}",
@@ -17,10 +18,19 @@
                "-xmlibdirname {build_dir}/xcelium.d"]
 
   run_opts:   ["-input {tool_srcs_dir}/xcelium_{dump}.tcl",
+               "-licqueue",
                "-64bit -xmlibdirname {build_dir}/xcelium.d -R",
                "+SVSEED={seed}",
                "+UVM_TESTNAME={uvm_test}",
                "+UVM_TEST_SEQ={uvm_test_seq}"]
+
+  // Vars that need to exported to the env.
+  exports: [
+    // Poll for an available license in all servers.
+    CDS_LIC_QUEUE_POLL: 1
+    // Poll for an available license every 1 min.
+    CDS_LIC_QUEUE_POLL_INT: 60
+  ]
 
   // Coverage related.
   // TODO: These options have to be filled in.

